vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -440,14 +463,20 @@ def predict_proba(self, data, as_pandas=True):
         """
         if self._classifier is None:
             raise RuntimeError('Classifier is not initialized, try `fit` first.')
-        proba = self._classifier.predict(data)
-        if 'image' in proba.columns:
-            ret = proba.groupby(["image"]).agg(list)
-        ret = proba
+        assert self._label_cleaner is not None
+        y_pred_proba = self._classifier.predict(data, with_proba=True)
+        if isinstance(data, pd.DataFrame) and 'image' in data:
```

Comment:
Don't fix in this PR, but long-term we don't want to be doing this kind of computation, as it is likely a significant increase to inference latency. I suspect changing gluoncv to output the ideal format in the first place might make things a lot simpler.